From 43457519028d99dec89e63ed792dda06069b9bd6 Mon Sep 17 00:00:00 2001
From: Ola Jeppsson <ola@adapteva.com>
Date: Fri, 28 Aug 2015 23:33:10 +0200
Subject: [PATCH 23/33] epiphany: Enable -mno-soft-cmpsf under -ffast-math

2015-08-28  Ola Jeppsson  <ola@adapteva.com>

	* gcc/config/epiphany/epiphany.h: Enable -mno-soft-cmpsf under
	-ffast-math.
	* gcc/doc/invoke.texi: Document.

Signed-off-by: Ola Jeppsson <ola@adapteva.com>
---
 gcc/ChangeLog.epiphany         | 6 ++++++
 gcc/config/epiphany/epiphany.h | 3 ++-
 gcc/doc/invoke.texi            | 3 ++-
 3 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/gcc/ChangeLog.epiphany b/gcc/ChangeLog.epiphany
index c17e2cbc39e..03ce4ce7375 100644
--- a/gcc/ChangeLog.epiphany
+++ b/gcc/ChangeLog.epiphany
@@ -1,3 +1,9 @@
+2015-08-28  Ola Jeppsson  <ola@adapteva.com>
+
+	* gcc/config/epiphany/epiphany.h: Enable -mno-soft-cmpsf under
+	-ffast-math.
+	* gcc/doc/invoke.texi: Document.
+
 2015-08-27  Ola Jeppsson  <ola@adapteva.com>
 
 	* config/epiphany/epiphany.h: Change USER_LABEL_PREFIX
diff --git a/gcc/config/epiphany/epiphany.h b/gcc/config/epiphany/epiphany.h
index 5d0c19de231..fc84cc40576 100644
--- a/gcc/config/epiphany/epiphany.h
+++ b/gcc/config/epiphany/epiphany.h
@@ -58,7 +58,8 @@ along with GCC; see the file COPYING3.  If not see
 #define ENDFILE_SPEC "crtend.o%s crtn.o%s " \
   "%{fsoftware-cache|fpic:cachemanager.o%s}"
 
-#define CC1_SPEC "%{fsoftware-cache:-fpic}"
+#define CC1_SPEC "%{fsoftware-cache:-fpic} " \
+  "%{ffast-math:-mno-soft-cmpsf}"
 
 #define EPIPHANY_LIBRARY_EXTRA_SPEC \
   "-ffixed-r40 -ffixed-r41 -ffixed-r42 -ffixed-r43"
diff --git a/gcc/doc/invoke.texi b/gcc/doc/invoke.texi
index 6d3cfcc664e..7cc7adfc202 100644
--- a/gcc/doc/invoke.texi
+++ b/gcc/doc/invoke.texi
@@ -17302,7 +17302,8 @@ and test the flags.  This is faster than a software comparison, but can
 get incorrect results in the presence of NaNs, or when two different small
 numbers are compared such that their difference is calculated as zero.
 The default is @option{-msoft-cmpsf}, which uses slower, but IEEE-compliant,
-software comparisons.
+software comparisons. @option{-mno-soft-cmpsf} is enabled under
+@option{-ffast-math}.
 
 @item -mstack-offset=@var{num}
 @opindex mstack-offset
-- 
2.31.1

